shell/platform/windows/compositor_software.cc

```diff
@@ -51,18 +51,105 @@ bool CompositorSoftware::Present(FlutterViewId view_id,
     return view->ClearSoftwareBitmap();
   }
 
-  // TODO: Support compositing layers and platform views.
-  // See: https://github.com/flutter/flutter/issues/31713
-  FML_DCHECK(layers_count == 1);
-  FML_DCHECK(layers[0]->offset.x == 0 && layers[0]->offset.y == 0);
-  FML_DCHECK(layers[0]->type == kFlutterLayerContentTypeBackingStore);
-  FML_DCHECK(layers[0]->backing_store->type ==
-             kFlutterBackingStoreTypeSoftware);
-
-  const auto& backing_store = layers[0]->backing_store->software;
-
-  return view->PresentSoftwareBitmap(
-      backing_store.allocation, backing_store.row_bytes, backing_store.height);
+  // Bypass composition logic if there is only one layer.
+  if (layers_count == 1) {
+    auto& layer = *layers[0];
+    if (layer.type == kFlutterLayerContentTypeBackingStore &&
+        layer.offset.x == 0 && layer.offset.y == 0) {
+      auto& backing_store = *layer.backing_store;
+      FML_DCHECK(backing_store.type == kFlutterBackingStoreTypeSoftware);
+      auto& software = backing_store.software;
+      return view->PresentSoftwareBitmap(software.allocation,
+                                         software.row_bytes, software.height);
+    }
+  }
+
+  // Composite many layers.
+  int x_min = INT_MAX;
+  int x_max = INT_MIN;
+  int y_min = INT_MAX;
+  int y_max = INT_MIN;
+  for (const FlutterLayer** layer = layers; layer < layers + layers_count;
+       layer++) {
+    auto& offset = (*layer)->offset;
+    auto& size = (*layer)->size;
+    x_min = std::min(x_min, static_cast<int>(offset.x));
+    y_min = std::min(y_min, static_cast<int>(offset.y));
```

Comment:
Consider adding a comment documenting why the conversion from double to int is safe for these.